platform/broadcom/sonic-platform-modules-mitac/ly1200-32x/opt/xcvr-serv/bms_xcvr_polld.c

```diff
@@ -0,0 +1,418 @@
+                    break;
+                case XCVR_RESET:
+                    ret_code = bms_xcvr_pin_write(xcvrirq_data[i].reset, val);
+                    break;
+                case XCVR_INTR_MSK:
+                    ret_code = bms_xcvr_pin_write(xcvrirq_data[i].intr_msk, val);
+                    break;
+                case XCVR_INTR_SRC:
+                    ret_code = bms_xcvr_pin_write(xcvrirq_data[i].intr_src, val);
+                    break;
+                case XCVR_LPMODE:
+                    ret_code = bms_xcvr_pin_write(xcvrirq_data[i].lpmode, val);
+                    break;
+                default:
+                    ret_code = -ENOTSUP;
+                    break;
+            }
+            if (ret_code < 0)
+                goto quit;
+        }
+    }
+quit:
+    return ret_code;
+}
+
+/*
+ * FEATURE:
+ *   bms_xcvr_pin_bmap_modify
+ * PURPOSE:
+ *   write bmap to file of sysfs if the value is modified
+ * PARAMETERS:
+ *   pin_num            (IN) the number means different pin name
+ *   value              (IN) new register value
+ *   old_value          (IN) last register value
+ * RETURNS:
+ *   open success, this function returns .
+ *   for other cases, negative errno is returned.
+ */
+int
+bms_xcvr_pin_bmap_modify(int pin_num,
+                         unsigned int value,
+                         unsigned int old_value)
+{
+    int ret_code = 0, i;
+    unsigned int val = 0;
+
+    for(i = 0; i < xcvrirq_data_num; i++) {
+        if (xcvrirq_data[i].present && (((old_value ^ value) >> i) & 0xff) ) {
+            val = (value >> i) & 0xff;
+            switch (pin_num) {
+                case XCVR_PRESENT:
+                    ret_code = bms_xcvr_pin_write(xcvrirq_data[i].present, val);
+                    break;
+                case XCVR_MODSEL:
+                    ret_code = bms_xcvr_pin_write(xcvrirq_data[i].modsel, val);
+                    break;
+                case XCVR_RESET:
+                    ret_code = bms_xcvr_pin_write(xcvrirq_data[i].reset, val);
+                    break;
+                case XCVR_INTR_MSK:
+                    ret_code = bms_xcvr_pin_write(xcvrirq_data[i].intr_msk, val);
+                    break;
+                case XCVR_INTR_SRC:
+                    ret_code = bms_xcvr_pin_write(xcvrirq_data[i].intr_src, val);
+                    break;
+                case XCVR_LPMODE:
+                    ret_code = bms_xcvr_pin_write(xcvrirq_data[i].lpmode, val);
+                    break;
+                default:
+                    ret_code = -ENOTSUP;
+                    break;
+            }
+            if (ret_code < 0)
+                goto quit;
+        }
+    }
+quit:
+    return ret_code;
+}
+
+/*
+ * FEATURE:
+ *   bms_xcvr_polld
+ * PURPOSE:
+ *   poll present pin for each port
+ * PARAMETERS:
+ *   args               (IN) NULL
+ * RETURNS:
+ *   No.
+ */
+void* bms_xcvr_polld(void *args)
```

Comment:
this is essentially the xrcvd does.

Reply:
If our xcvr-servd is not running in SONiC, we will not see "state UP" via the "ip addr" command and will not see anything through the "show interfaces transceiver eeprom" command when plugging in the transceiver.

The implementation difference between our xcvr-servd and SONiC's xrcvd might be related to our hardware design. Please allow our xcvr-servd to run on SONiC for our platform module.